src/cljam/io/vcf/util.clj

```diff
@@ -286,3 +286,93 @@
                       (if before? \[ \]))]
         (str (when-not before? s) bracket chr \: pos bracket (when before? s)))
       (str (when (= :before join) \.) s (when (= :after join) \.)))))
+
+(defn- inspect-nucleotides-allele
+  [ref alt]
+  (let [ref-length (count ref)
+        alt-length (count alt)
+        upper-ref (cstr/upper-case ref)
+        upper-alt (cstr/upper-case alt)
+        left-match (->> (map = upper-ref upper-alt)
+                        (take-while true?)
+                        count)
+        right-match (->> (cstr/reverse upper-alt)
+                         (map = (cstr/reverse upper-ref))
+                         (take-while true?)
+                         count
+                         long
+                         (Math/min (- (Math/min ref-length alt-length)
+                                      left-match)))
+        matched-length (+ left-match right-match)]
+    ;; Note that `{:type :ref}` is handled in the caller
+    (cond
+      (and (= left-match ref-length) (< left-match alt-length))
+      {:type :insertion, :offset (dec left-match),
+       :n-bases (- alt-length left-match), :inserted (subs alt left-match)}
+
+      (and (< left-match ref-length) (= left-match alt-length))
+      {:type :deletion, :offset (dec left-match),
+       :n-bases (- left-match ref-length), :deleted (subs ref left-match)}
```

Comment:
I think `:n-bases` means the number of inserted/deleted bases, but it's bounded minus value (i.e. non-natural number) when `:type` is `:deletion`. Maybe it's more appropriate to make `:n-bases` take only natural numbers.

Reply:
Thanks! Fixed in 07aa4d7